PM-2749 full-width preference persist on sso login

Jingo88 opened the pull request:
```
- [ X ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This update ensures the users preference for `display full width` will persist when logging in via SSO. 

## Code changes

Single line added to `lock.component.ts` inside web. This will call the `messagingService` with `setFullWidth` that will invoke the `broadcasterService.subscribe` logic inside the `app.component`

## Video 

The below video includes updating the display preferences through both Master Password Login as well as SSO and watching that preference persist when we log into the vault in new tabs.


https://github.com/bitwarden/clients/assets/8302660/56566595-d6b7-4842-bd62-1c0e42d2f6ba


Updated More Video

SSO + 2FA

https://github.com/bitwarden/clients/assets/8302660/ca8a9f56-88fa-4f43-985d-e4f654a7d10a



SSO + Admin Request / Device Request


https://github.com/bitwarden/clients/assets/8302660/163f1ac3-6bac-4e0f-8115-c7327909d6e1



Standard Login With Device + Verification Code


https://github.com/bitwarden/clients/assets/8302660/51a0c2e7-0379-4f3c-ab4a-856b15198379

Jingo88 committed:
updated lock component so full-width preference persist on sso login

gbubemismith reviewed (commented):
I wonder why this didn't exist on the `lock component` cause you can add `onSuccessfulLogin` on the `sso component` and have the same effect.  @differsthecat, what do you think?

differsthecat commented:
> I wonder why this didn't exist on the `lock component` cause you can add `onSuccessfulLogin` on the `sso component` and have the same effect. @differsthecat, what do you think?

If I were to guess, it was an oversight at some point. There are many login flows and it's easy to miss one.

gbubemismith reviewed (dismissed):
Looks good. Nice job @Jingo88

trmartin4 commented:
Do we also need to add this to the `two-factor.component.ts`, `login-with-device.component.ts`, and `sso.component.ts`?  Those can also both land the user on the vault.

Jingo88 committed:
Merge branch 'master' into PM-2749-full-width-persist-on-sso

Jingo88 committed:
Merge branch 'master' into PM-2749-full-width-persist-on-sso

jlf0dev reviewed (commented):
Looks good but I am also curious about Todd's question ^

Jingo88 committed:
added code in login-with-device in web for set-width messaging service send

Jingo88 commented:
Apologies, I wasn't aware of the other flows. I've added similar logic to the `login-with-device.component` inside web. The PR description has been updated with extra videos for SSO with 2FA, Admin Request, Trusted Device.

JaredSnider-Bitwarden commented on apps/web/src/app/auth/login/login-with-device.component.ts:
```diff
@@ -67,4 +70,12 @@ export class LoginWithDeviceComponent
     );
   }
+
+  async handleApprovedAdminAuthRequest(
```
⚠️ We should be overriding the base `onSuccessfulLogin()` callback instead of the `handleApprovedAdminAuthRequest(...)`method.  

This won't work for the standard, non SSO [login with device](https://bitwarden.com/help/log-in-with-device/) flow as currently written.

JaredSnider-Bitwarden replied:
Aside: really appreciate all the videos showing the testing of the various flows! 🏅

Jingo88 committed:
updated login-with-device to call super.onSuccessfulLogin

Jingo88 commented:
Updated `login-with-device` to target the `onSuccessfulLogin` callback. All previous flows in videos working as intended. Added an additional video at the end of PR description for `Standard login with device + Verification Code`

JaredSnider-Bitwarden reviewed (approved):
LGTM!

Jingo88 commented:
No longer needed with Vertical Nav work
